portfolio/src/main/webapp/script.js

```diff
@@ -47,8 +47,31 @@ fetch("footer.html")
                 });
 
 //fetching the welcome message
-function fetchWelcome() {
+/*function fetchWelcome() {
   fetch('/data').then(response => response.text()).then((message) => {
     document.getElementById('welcome-container').innerText = message;
   });
+}*/
+
+//fetching the JSON arraylist string from server
+function fetchList() {
+  fetch('/data').then(response => response.json()).then((citiesList) => {
+    console.log(citiesList);
+    const citiesListElement = document.getElementById('cities-list-container');
+    citiesListElement.innerHTML = '';
+    citiesListElement.appendChild(
```

Comment:
this should be done in a loop, there is no guarantee you will get 3 items back.

Reply:
Ok, I have implemented a For Each loop instead!